service/history/workflow/delete_manager.go

```diff
@@ -95,16 +95,25 @@ func (m *DeleteManagerImpl) AddDeleteWorkflowExecutionTask(
 	visibilityQueueAckLevel int64,
 ) error {
 
-	// Create delete workflow execution task only if workflow is closed successfully and all pending tasks are completed (if in active cluster).
-	// Otherwise, mutable state might be deleted before close tasks are executed.
-	// Unfortunately, queue ack levels are updated with delay (default 30s),
-	// therefore this API will return error if workflow is deleted within 30 seconds after close.
-	// The check is on API call side, not on task processor side because visibility delete task doesn't have access to mutable state.
+	// Create `DeleteWorkflowExecutionTask` only if workflow is closed successfully and all pending tasks are completed (if in active cluster).
+	// Otherwise, mutable state might be deleted before close tasks are executed (race condition between close and delete tasks).
+	// Unfortunately, queue ack levels are updated with delay (default 60s),
+	// therefore this API will return error if workflow is deleted within 60 seconds after close.
```

Comment:
Why 60s? For transfer queue, cluster ack level is updated every 30s, for visibility queue, overall ack level is also updated every 30s (since there's only one queue for visibility).

Reply:
:facepalm: